agent/discovery/host.go

```diff
@@ -22,18 +23,22 @@ type HostDiscovery struct {
 	discovery  BaseDiscovery
 }
 
-func NewHostDiscovery(sshAddress string, collectorClient collector.Client) HostDiscovery {
+func NewHostDiscovery(sshAddress string, collectorClient collector.Client, interval time.Duration) HostDiscovery {
 	d := HostDiscovery{}
 	d.id = HostDiscoveryId
 	d.sshAddress = sshAddress
-	d.discovery = NewDiscovery(collectorClient)
+	d.discovery = NewDiscovery(collectorClient, interval)
 	return d
 }
 
 func (h HostDiscovery) GetId() string {
 	return h.id
 }
 
+func (d HostDiscovery) GetInterval() time.Duration {
```

Comment:
Is this needed? Can we use the one in `BaseDiscovery`? :thinking:

Reply:
I don't think it is, as of now there is no reason to not use the one in BaseDiscovery, I have no strong opinion here; it's just a matter of taste as the attribute is really stored in the BaseDiscovery

Reply:
@arbulu89 not sure, it feels a bit wrong to have exactly the same attribute (`interval`) repeated in the `HostDiscovery`, `ClusterDiscovery`,.... but I see your point about the `BaseDiscovery` not being that useful. I'd address that in a separate PR (I'll create an issue if you agree with this) and just move the `GetInterval()` function there for now. WDYT?

Reply:
I ended up improving this a bit. Now it makes a bit more sense as we don't have the BaseDiscovery struct anymore but the intervals are stored in each Discovery that implements the Discovery interface, but of course the GetInterval() is still there. I think it has sense to be there so that at some point we can e.g. query the effective interval values to display them somewhere.